x-pack/plugins/canvas/public/state/middleware/es_persist.js

```diff
@@ -43,6 +44,8 @@ export const esPersistMiddleware = ({ getState }) => {
     next(action);
     const newState = getState();
 
+    if (getReadOnly(newState)) return;
```

Comment:
@rashidkpc I tried moving the current page from being stored as part of the workpad to transient state, but there were conflicts with the aeroelastic store where it relies the selected page to stay in the workpad state. I talked to @w33ble about the issues I was encountering, and we decided moving the selected page to transient state may be easier to implement after refactoring our redux implementation.

This was the simpler fix for preventing the user from updating the current page in the saved workpad object in elasticsearch.